frontend/src/tests/application-tests.js

```diff
@@ -1,24 +1,428 @@
+        hours: 68,
+    };
+
+    const userWithTaPermissions = {
+        roles: ["ta"],
+        utorid: "greenb",
+    };
+
+    /**
+     * Switches the current active user to the seededData.applicant in the seeder.
+     * This function uses debug route to achieve user switching.
+     *
+     * @returns {Promise<void>}
+     */
+    async function switchToApplicantUser() {
+        let resp = await apiPOST(
+            `/debug/active_user`,
+            userCreatedFromApplicant
+        );
+        expect(resp).toHaveStatus("success");
+    }
+
+    /**
+     * Switches the current active user to a user with only the ta role.
+     * This function uses debug route to achieve user switching.
+     *
+     * @returns {Promise<void>}
+     */
+    async function switchToTaUser() {
+        let resp = await apiPOST(`/debug/active_user`, userWithTaPermissions);
+        expect(resp).toHaveStatus("success");
+    }
+
+    /**
+     * Restores the active user to the default user (the user logged during test setup in beforeAll).
+     *
+     * @returns {Promise<void>}
+     */
+    async function restoreDefaultUser() {
+        let respSwitchBackUser = await apiPOST(
+            `/debug/active_user`,
+            defaultUser
+        );
+        expect(respSwitchBackUser).toHaveStatus("success");
+    }
+
     beforeAll(async () => {
         await databaseSeeder.seed({ apiGET, apiPOST });
+        session = databaseSeeder.seededData.session;
+        applicant = databaseSeeder.seededData.applicant;
+        position = databaseSeeder.seededData.position;
+
+        // Get the current active user
+        let resp = await apiGET("/debug/active_user");
+        expect(resp).toHaveStatus("success");
+        defaultUser = resp.payload;
+
+        // Making sure the user is the seeded applicant before this get request for the purposes of the next test
+        // Created a user for the applicant to make sure they exist before changing the default user to applicant
+        Object.assign(userCreatedFromApplicant, {
+            utorid: applicant.utorid,
+            roles: ["ta"],
+        });
+
+        resp = await apiPOST("/debug/users", userCreatedFromApplicant);
+        expect(resp).toHaveStatus("success");
+        Object.assign(userCreatedFromApplicant, resp.payload);
+
+        // Create a new user with TA permissions
+        resp = await apiPOST("/debug/users", userWithTaPermissions);
+        expect(resp).toHaveStatus("success");
+        Object.assign(userWithTaPermissions, resp.payload);
     }, 30000);
 
     // These tests set data through the `/public/postings` route,
     // but read data through the `/api/v1/admin` route.
     describe("Public route tests", () => {
-        it.todo("Get survey.js posting data through public route");
-        it.todo(
-            "Survey.js posting data is pre-filled based on prior applicant/application data"
-        );
+        it("Get survey.js posting data through public route", async () => {
```

Comment:
The description of this test and its content don't seem to match...I see nothing dealing with a public route.

Reply:
See line 116

Reply:
I'm still confused. Can you explain?

Reply:
Oh I understand the confusion. I have mixed up this test with the next one (oops) and I now understand what you mean. I have added the important part that does what the test description implies. 

Also I realized that I shouldn't be referencing lune numbers since they change every commit. Won't happen again 😅